include/deal.II/grid/tria.h

```diff
@@ -3482,6 +3482,34 @@ class Triangulation : public Subscriptor
     std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3>>> &
   get_periodic_face_map() const;
 
+  /**
+   * Translate the unique id of a coarse cell to its index.
+   *
+   * @note: For serial and shared triangulation both id and index are the same.
+   *        For distributed triangulations setting both might differ, since the
+   *        id might correspond to a global id and the index a local id.
+   *
+   * @param coarse_cell_id Unique id of the coarse cell.
+   * @return Index of the coarse cell.
+   */
+  virtual unsigned int
+  coarse_cell_id_to_coarse_cell_index(
+    const unsigned long int coarse_cell_id) const;
```

Comment:
Only one of these two would then be `types::coarse_cell_id`, and similarly for the function below.

Reply:
Thanks @bangerth! This is what I was asking above. I have modified the code accordingly. I still have to add some comments.